language-codes-are-now-normalized.rst

```diff
@@ -0,0 +1,36 @@
+.. post:: October 24, 2023
+   :tags: docs, languages, translations
+   :category: Changelog
+   :author: Santos
+   :location: CUE
+
+Language codes are now normalized
+=================================
+
+The following language codes are now normalized to be lowercase and use a dash as a separator instead of an underscore:
+
+- ``nb_NO`` is now ``nb-no``
+- ``pt_BR`` is now ``pt-br``
+- ``es_MX`` is now ``es-mx``
+- ``uk_UA`` is now ``uk-ua``
+- ``zh_CN`` is now ``zh-cn``
+- ``zh_TW`` is now ``zh-tw``
+
+This change affects the following areas:
+
+API:
+   - When creating or updating projects, you must now use the new normalized language codes.
+   - The API responses will include the new normalized language codes.
+
+Docs and download URLs:
+   Previous URLs from projects using the old language codes will redirect to the new normalized language codes.
+   For example:
+
+   - ``https://docs.readthedocs/pt_BR/latest/`` will redirect to ``https://docs.readthedocs/pt-br/latest/``.
+   - ``https://docs.readthedocs/_/downloads/pt_BR/latest/pdf/`` will redirect to ``https://docs.readthedocs/_/downloads/pt-br/latest/pdf/``.
+
+Redirects:
+  If you had redirects using the old language codes in the ``From URL`` field,
+  they were automatically updated to use the new normalized language codes.
```

Comment:
Just realized about this, looks like we only need to worry about them in .org, we don't have any redirects using the old language codes in .com.

```
old_language_codes = [
    "nb_NO",
    "pt_BR",
    "es_MX",
    "uk_UA",
    "zh_CN",
    "zh_TW",
]

for r in Redirect.objects.filter(redirect_type='exact'):
    for lang in old_language_codes:
        if lang in r.from_url:
            print(r.id, r.project.slug, r.from_url, '->', r.to_url)
```

We have 27 redirects to update on .org only. I can do that after deploy.

Reply:
Easy enough 👍